src/main/java/com/example/demo/controller/ProductController.java

```diff
@@ -1,54 +1,62 @@
 package com.example.demo.controller;
 
+import com.example.demo.dto.ProductDto;
 import com.example.demo.model.Product;
 import com.example.demo.service.ProductService;
-import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
-import org.springframework.web.bind.annotation.DeleteMapping;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.PutMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
 @Controller
 @RequestMapping("/products")
-@RequiredArgsConstructor
 public class ProductController {
     private final ProductService productService;
 
+    @Autowired
+    public ProductController(ProductService productService) {
+        this.productService = productService;
+    }
+
     @PostMapping
-    public ResponseEntity<Product> createProduct(@RequestBody Product product) {
+    public ResponseEntity<ProductDto> createProduct(@RequestBody ProductDto productDto) {
+        Product product = mapToProduct(productDto);
         Product createdProduct = productService.createProduct(product);
-        return ResponseEntity.ok(createdProduct);
+        ProductDto createdProductDto = mapToProductDto(createdProduct);
+        return ResponseEntity.ok(createdProductDto);
     }
 
     @GetMapping
     public String getAllProducts(Model model) {
         List<Product> productList = productService.getAllProducts();
-        model.addAttribute("products", productList);
+        List<ProductDto> productDtoList = productList.stream()
+                .map(this::mapToProductDto)
```

Comment:
Hopefully you're aware of how the operation used here works, otherwise I recommend reading up on Lambda and Streams in Java

Reply:
yip, putting references in the code was intentional - I don't like lambda so I try not to use it when I don't have to.
but thanks for pointing it out, i'll try to do it the other way as well.